src/script/lua_api/l_env.cpp

```diff
@@ -970,16 +970,16 @@ int ModApiEnvBase::findNodesInArea(lua_State *L, const NodeDefManager *ndef,
 		});
 
 		// last filter table is at top of stack
-		u32 i = filter.size() - 1;
-		do {
+		s32 i = filter.size();
+		while (i --> 0) {
```

Comment:
shouldn't `i-- != 0` work without needing to change the type?

Reply:
You're right. I in fact tested the first version with `u32` and - for whatever reason - thought that I need to have it signed if `i = 0`, which is not true.

Reply:
You're right. I in fact tested the first version with `u32` and - for whatever reason - thought that I need to have it signed if `i = 0`, which is not necessary.